service/frontend/protojson_marshaler.go

```diff
@@ -30,39 +30,77 @@ import (
 	"io"
 
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
-	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/proto"
+
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/api/temporalproto"
+)
+
+const (
+	mimeJSON                  = "application/json"
```

Comment:
```suggestion
	mimeAll                   = "*"
```

I think we should support unset mime type to be JSON too. I just did this simply as "any unrecognized mime type" but if you want to be more strict when user does specify it, I suppose we can though probably no need.